core/dbt/contracts/results.py

```diff
@@ -268,10 +271,39 @@ def from_execution_results(
         )
         return cls(metadata=meta, results=processed_results, elapsed_time=elapsed_time, args=args)
 
+    @classmethod
+    def compatible_previous_versions(cls) -> Iterable[Tuple[str, int]]:
+        return [
+            ("run-results", 4),
+        ]
+
+    @classmethod
+    def upgrade_schema_version(cls, data):
+        """This overrides the "upgrade_schema_version" call in VersionedSchema (via
+        ArtifactMixin) to modify the dictionary passed in from earlier versions of the run_results."""
+        run_results_schema_version = get_artifact_schema_version(data)
+        # If less than the current version (v5), preprocess contents to match latest schema version
+        if run_results_schema_version <= 5:
+            data = upgrade_run_results_json(data, run_results_schema_version)
+        return cls.from_dict(data)
+
     def write(self, path: str):
         write_json(path, self.to_dict(omit_none=False))
 
 
+def upgrade_run_results_json(run_results: dict, run_results_schema_version: int) -> dict:
+    # In v5, we added 'compiled' attributes to each result entry
+    # Going forward, dbt expects these to be populated
+    if run_results_schema_version <= 5:
+        for result in run_results["results"]:
+            result["compiled"] = False
+            result["compiled_code"] = ""
+            result["relation_name"] = ""
+            result["depends_on"] = {}
```

Comment:
I added this here because it had been mentioned in https://github.com/dbt-labs/dbt-core/issues/7519, but it's not actually part of [run results v5](https://schemas.getdbt.com/dbt/run-results/v5.json), so we should remove it